### PR TITLE
[DOCS] Reformat delimited payload token filter docs

### PR DESCRIPTION
**Pull request by jrodewig**
Reformats the delimted payload token filter docs as part of #44726.

### Changes
* Adds a title abbreviation
* Relocates the older name deprecation warning
* Updates the description and adds a Lucene link
* Adds a note to explain payloads and how to store them
* Adds analyze and custom analyzer snippets
* Adds a 'Return stored payloads' example

@mayya-sharipova @jtibshirani  Would you mind looking at this at your convenience? I want to ensure the added information about payloads and term vectors works. Thanks!

### CODE REVIEW

File: docs/reference/analysis/tokenfilters/delimited-payload-tokenfilter.asciidoc [hunk 1]
```diff
@@ -1,21 +1,331 @@
...
+
+This filter uses Lucene's
+https://lucene.apache.org/core/{lucene_version_path}/analyzers-common/org/apache/lucene/analysis/payloads/DelimitedPayloadTokenFilter.html[DelimitedPayloadTokenFilter].
+
+[NOTE]
+.Payloads
+====
+A payload is user-defined binary data associated with a token position and
+stored as base64-encoded bytes. Payloads are often used with the
+<<query-dsl-script-score-query,`script_score`>> query to calculate custom scores
```
Comment (mayya-sharipova) [inline docs/reference/analysis/tokenfilters/delimited-payload-tokenfilter.asciidoc]: I guess would be nice to do this some time, but currently there is no way for `script_score` to access payloads.  I don't know any other ES query that can deal with payloads either. The only way to access them is through _termvectors API.

File: docs/reference/analysis/tokenfilters/delimited-payload-tokenfilter.asciidoc [hunk 2]
```diff
@@ -1,21 +1,331 @@
...
+{es} does not store token payloads by default. To store payloads, you must:
+
+
+* Set mapping parameters as follows for any field storing payloads:
+** <<mapping-store,`store`>> to `true`
+** <<term-vector,`term_vector`>> to `with_positions_payloads` or
+   `with_positions_offsets_payloads`
+
+* Use an index analyzer that includes the `delimted_payload` filter
+====
```
Comment (mayya-sharipova) [inline docs/reference/analysis/tokenfilters/delimited-payload-tokenfilter.asciidoc]: I am not sure about the correctness of this paragraph.  So, for a text/keyword field ES can create 3 Lucene fields:  1) usual indexed field broken into terms and used for search. Here we don't sore payloads (I think) and definitely never use them in any search query. 2) a stored field with index option `store:true`.  But I am not sure why you mentioned it here (it seems to me that it doesn't have to do anything with payloads).  3) a term vectors field with index option of `term_vector`. these are primarily used for highlighting, but can be used just for retrieval purpose as well (as in your examples). And here we can store payloads but again we don't use payloads for anything intelligent besides just retrieval. 

nit: `delimted_payload` -> `delimited_payload`

File: docs/reference/analysis/tokenfilters/delimited-payload-tokenfilter.asciidoc [hunk 3]
```diff
@@ -1,21 +1,331 @@
...
+}
+--------------------------------------------------
+
+[[analysis-delimited-payload-tokenfilter-return-stored-payloads]]
+==== Return stored payloads
+
+Use the <<indices-create-index,create index API>> to create an index that:
+
+* Includes a field that stores payloads. For this field, set the
+  <<mapping-store,`store`>> mapping parameter to `true` and the
```
Comment (mayya-sharipova) [inline docs/reference/analysis/tokenfilters/delimited-payload-tokenfilter.asciidoc]: I am not sure why `store` : `true` is necessary? 

Includes a field that stores payloads -> I would reformulate it to something like "stores term vectors with payloads", as it not a usual indexed field with payloads.

**Review by mayya-sharipova — CHANGES_REQUESTED** [review]
@jrodewig Thanks, nice work, but needs some changes.

File: docs/reference/analysis/tokenfilters/delimited-payload-tokenfilter.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/analysis/tokenfilters/delimited-payload-tokenfilter.asciidoc]: Thank you for catching this. I've updated this note to state that you can view stored payloads using the term vectors API API.

File: docs/reference/analysis/tokenfilters/delimited-payload-tokenfilter.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/analysis/tokenfilters/delimited-payload-tokenfilter.asciidoc]: Thanks for another great catch. I've removed references to the `store` requirement throughout. I also fixed the typo.

File: docs/reference/analysis/tokenfilters/delimited-payload-tokenfilter.asciidoc (hunk 3 above)
Comment (jrodewig) [reply docs/reference/analysis/tokenfilters/delimited-payload-tokenfilter.asciidoc]: It's not. I've rephrased this bullet to "Includes a field that stores term vectors with payloads." as you suggested. Thanks!

**Comment by jrodewig** [comment]
Thanks for your thorough review, @mayya-sharipova. You cleared up some of my misunderstandings around storing payloads and their use cases.

This PR is ready for another look at your convenience. Thanks again!

**Review by mayya-sharipova — APPROVED** [review]
Thanks @jrodewig, the changes LGTM

**Review by mayya-sharipova — APPROVED** [review]
Thanks @jrodewig, the changes LGTM

**Pull request merged**